src/events/tls.py

```diff
@@ -116,6 +118,11 @@ def _on_certificate_expiring(self, _: EventBase) -> None:
 
         self.charm.state.unit_server.update({"csr": new_csr.decode("utf-8").strip()})
 
+    def _on_config_changed(self, event: EventBase):
+        """If system configuration (such as IP) changes, certs have to be re-issued."""
+        if self.charm.state.unit_server.tls and not self.charm.tls_manager.certificate_valid():
```

Comment:
Isn't that a sufficient indicator that the IP in the currently stored certificate is different from the current IP of the node...?

Reply:
..at the moment when receiving a `config-changed` event?

(Which only triggers on 3 distinct cases: Actual configuration change, right after startup, AND on IP change...?
https://ops.readthedocs.io/en/latest/#ops.ConfigChangedEvent)

I'm asking because it's a non-trivial change, and except if strong beenfid, I'd prefer to skip it.

HA tests have been pretty elaborate.

Reply:
Moreover. Config check shouldn't go int the TLS event handlers. 

From TLS perspective we have one duty to do on a `config-changed` eevnt. We have to check if the certificates are still relevant -- and if they are not, we must request new ones. That is the scope of the TLS `config-changed` event handler.

Now note that on `config-changed` events the main `reconcile()` method is being also triggered. In case the config is to be changed (or may have changed) that's the execution to be triggered.

Overall: your concern regarding the config is valid, however it's not the right place (and time) yet to worry about that one here :-)

Reply:
This is tracked here. https://github.com/canonical/opensearch-dashboards-operator/issues/27

Reply:
Issue closed as I'm implementing it already at the moment.

Reply:
Responded.